src/software/world/ball.h

```diff
@@ -6,43 +6,48 @@
 
 #include "software/geom/point.h"
 #include "software/util/time/timestamp.h"
+#include "software/world/ball_state.h"
 
 class Ball final
 {
    public:
     /**
-     * Creates a new ball with the given position and velocity
+     * Creates a new ball with a new state given by the position and velocity
      *
      * @param position The position of the ball, with coordinates in metres
      * @param velocity The velocity of the ball, in metres per second
      * @param timestamp The timestamp at which the ball was observed to be at the
      * given position and velocity
-     * @param history_duration The number of previous ball states that should be stored.
+     * @param history_size The number of previous ball states that should be stored.
      *
      */
-    explicit Ball(Point position, Vector velocity, const Timestamp& timestamp,
-                  unsigned int history_duration = 20);
+    explicit Ball(Point position, Vector velocity, const Timestamp &timestamp,
+                  unsigned int history_size = 20);
+
 
     /**
-     * Updates the ball with new data, updating the current data as well as the predictive
-     * model
+     * Creates a new ball with the given BallState
      *
-     * @param new_position the new position of the ball, defined in metres
-     * @param new_velocity the new velocity of the ball, defined in metres per second
-     * @param timestamp The timestamp at which the ball was observed to be at the given
-     * position and velocity. The timestamp must be >= the ball's latest update timestamp
+     * @param ball_state the state of the ball
+     * @param history_size the number of previous ball states that should be stored
      */
-    void updateState(const Point& new_position, const Vector& new_velocity,
-                     const Timestamp& timestamp);
+    explicit Ball(BallState &ball_state, unsigned int history_size = 20);
+
+    /**
+     * Returns the last updated BallState of the ball
```

Comment:
nit: Although kinda repetitive, I think something like "Returns the current state of the ball" is simpler and more clear.

Reply:
updated